inc/theme-hooks.php

```diff
@@ -125,3 +125,123 @@ function alcatraz_output_footer_bottom() {
+
+	global $post;
+
+	$args = array();
+
+	//defaults
+	$defaults = array(
+		'show_all'         => true,
+		'show_on_home'     => false,
+		'show_empty'       => false,
+		'sort_by'          => 'menu_order',
+		'a_heading'        => false,
+		'title'            => ''
+	);
+
+	$args = wp_parse_args( (array)$args, $defaults );
+
+	// Get clean param values.
+	$show_all         = $args['show_all'];
+	$show_on_home     = $args['show_on_home'];
+	$show_empty       = $args['show_empty'];
+	$sort_by          = $args['sort_by'];
+	$a_heading        = $args['a_heading'];
+	$title            = $args['title'];
+
+	if ( is_search() || is_404() ) {
+		return false; //doesn't apply to search or 404 page
+	}
+
+	if ( is_front_page() ) {
+		return false;	//if we're on the front page and we haven't chosen to show this anyways, leave
+	}
+
+	if(!$post->post_parent) {
+		$children = wp_list_pages("title_li=&child_of=".$post->ID."&echo=0");
+		}else{
+		if($post->ancestors){
+			$ancestors = end($post->ancestors);
+	 	}
+	 }
+
+	if ( is_page() ) {
+
+		if ( isset( $post) && is_object( $post ) ) {
+		get_post_ancestors($post);
+   		//workaround for occassional problems
+		} else {
+
+			if ($post_page = get_option("page_for_posts")) {
+
+		 		$post = get_page($post_page);
+			}
+		 //treat the posts page as the current page if applicable
+		else return false;
+
+		}
+
+		if ( is_front_page() || isset( $sub_front_page ) ) {
+
+			echo "<ul>";
+
+			$children = wp_list_pages(array(
+				'title_li'    => '',
+				'depth'       => 1,
+				'sort_column' => $sort_by,
+				'echo'        => false
+				));
+
+			echo $children;
+			echo "</ul>";
+			return true;
+	  	}
+
+		$post_ancestors = ( isset($post->ancestors) ) ? $post->ancestors : get_post_ancestors($post); //get the current page's ancestors either from existing value or by executing function
+		$top_page = $post_ancestors ? end($post_ancestors) : $post->ID; //get the top page id
+
+		$thedepth = 0; //initialize default variables
+
+		$excluded = explode(',', $exclude_list); //convert list of excluded pages to array
+
+		if ( in_array( $post->ID,$excluded ) ) {
+			return false; //if on excluded page, and setup to hide on excluded pages
+		}
+
+		$post_ancestors = ( isset($post->ancestors) ) ? $post->ancestors : get_post_ancestors($post); //get the current page's ancestors either from existing value or by executing function
+		$top_page = $post_ancestors ? end($post_ancestors) : $post->ID; //get the top page id
+
+		$children = wp_list_pages(array(
+		'title_li'    => '',
+		'echo'        => 0,
+		'depth'       => $thedepth,
+		'child_of'    => $top_page,
+		'sort_column' => $sort_by
+		 ));	//get the list of pages, including only those in our page list
+
+		if( ! $children && ! $show_empty) {
+			return false; 	//if there are no pages in this section, and user hasnt chosen to display widget anyways, leave the function
+		}
+
+		echo "<div id='secondary' class='primary-sidebar sidebar' role='complementary'>";
```

Comment:
Let's invert these quotes so that we use single quotes to wrap strings generally and then double quotes to wrap HTML attributes. If there are any cases where we're dealing with variables in the string then we can either break out of the string or use double quotes in those situations.